### PR TITLE
Scheduler

### PR DESCRIPTION
**Pull request by LinaSabboula**

### CODE REVIEW

File: app/Console/Kernel.php [hunk 1]
```diff
@@ -24,8 +24,9 @@ class Kernel extends ConsoleKernel
      */
     protected function schedule(Schedule $schedule)
     {
-        // $schedule->command('inspire')->hourly();
+//         $schedule->command('inspire')->everyMinute();
         $schedule->command('telescope:prune')->daily();
+        $schedule->call('App\Http\Controllers\GovernmentController@activateGovernments')->hourly();
```
Comment (Mhany95) [inline app/Console/Kernel.php]: Use a job instead of a direct call to a method

File: app/Console/Kernel.php [hunk 2]
```diff
@@ -24,8 +25,9 @@ class Kernel extends ConsoleKernel
      */
     protected function schedule(Schedule $schedule)
     {
-        // $schedule->command('inspire')->hourly();
+//         $schedule->command('inspire')->everyMinute();
         $schedule->command('telescope:prune')->daily();
+        $schedule->job(new DeactivateGovernments)->hourly();
```
Comment (Mhany95) [inline app/Console/Kernel.php]: Add the job to a command and run the command here in the scheduler

File: app/Jobs/DeactivateGovernments.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
```
Comment (Mhany95) [inline app/Jobs/DeactivateGovernments.php]: Set a queue

File: app/Jobs/DeactivateGovernments.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        Government::doesnthave('cities')
```
Comment (Mhany95) [inline app/Jobs/DeactivateGovernments.php]: Logic should be moved to a controller

File: app/Nova/Government.php [hunk 5]
```diff
@@ -48,6 +49,7 @@ public function fields(Request $request)
         return [
             ID::make(__('ID'), 'id')->sortable(),
             Text::make('Name', 'name')->sortable(),
+            Boolean::make('Active', 'is_active')->sortable(),
```
Comment (Mhany95) [inline app/Nova/Government.php:52]: Missing reference in model

File: database/migrations/2021_05_08_123009_create_governments_table.php [hunk 6]
```diff
@@ -16,6 +16,7 @@ public function up()
         Schema::create('governments', function (Blueprint $table) {
             $table->id();
             $table->string('name')->unique();
+            $table->boolean('is_active')->default(false);
```
Comment (Mhany95) [inline database/migrations/2021_05_08_123009_create_governments_table.php]: This is incorrect, you are now on production, you can't run a fresh migration

File: app/Console/Commands/DeactivateGovernments.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     */
+    public function handle()
+    {
+        \App\Jobs\DeactivateGovernments::dispatch();
```
Comment (Mhany95) [inline app/Console/Commands/DeactivateGovernments.php]: Import the job

File: app/Models/InactiveGovernment.php [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class InactiveGovernment extends Model
+{
+    use HasFactory;
+    protected $fillable = [
+        'government_id',
+    ];
+    public function government(){
```
Comment (Mhany95) [inline app/Models/InactiveGovernment.php:14]: 1- Missing function in gov model
2- Add nova resource

File: database/migrations/2021_06_01_153459_add_is_active_to_governments_table.php [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('governments', function (Blueprint $table) {
+                $table->dropColumn('is_active');
```
Comment (abdelrahmanna) [inline database/migrations/2021_06_01_153459_add_is_active_to_governments_table.php:30]: check if the col exists before dropping it

**Pull request merged**